src/main/java/org/springframework/data/elasticsearch/client/TransportClientFactoryBean.java

```diff
@@ -86,15 +88,20 @@ public void afterPropertiesSet() throws Exception {
 	}
 
 	protected void buildClient() throws Exception {
-		client = TransportClient.builder().settings(settings()).build();
+		client = TransportClient.builder().addPlugin(ShieldPlugin.class).settings(settings()).build();

```

Comment:
IMO this should be optional.
E.g. if there is a bean of type `List<Class<elasticsearch.Plugin>>` in the app context than for each item in the list call `.addPlugin(item)`